superset/utils/date_parser.py

```diff
@@ -106,6 +119,16 @@ def dttm_from_timetuple(date_: struct_time) -> datetime:
     )
 
 
+def get_past_or_future(
+    human_readable: Optional[str], source_time: Optional[datetime] = None,
+) -> datetime:
+    cal = parsedatetime.Calendar()
```

Comment:
The next few lines have a lot in common with `parse_human_timedelta`, should we refactor them both use a common simple method?

Reply:
This function returns `datetime`, while the following ones return `timedelta`.

Reply:
I will open a separate PR to fix these problems.